packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,252 @@
+
+export const handleDragEnd = (
+  dragResult,
+  onUpdateColumns,
+  selectedColumns,
+  availableColumns,
+  setIsDragging
+) => {
+  setIsDragging(false);
+  // Invalid drop destination, do nothing
+  if (!dragResult.destination) return;
+  if (dragResult.destination.droppableId === HIDDEN_COLUMNS_PANEL) {
+    if (dragResult.source.droppableId === HIDDEN_COLUMNS_PANEL) return;
+    onUpdateColumns([
+      ...selectedColumns.slice(0, dragResult.source.index),
+      ...selectedColumns.slice(dragResult.source.index + 1),
+    ]);
+  } else {
+    // the destination is the selected columns panel
+
+    // it's a swap when the source and the destination are the same
+    const isSwap = dragResult.source.droppableId === SELECTED_COLUMNS_PANEL;
+
+    const items = isSwap
+      ? // remove the dragged item from its position if it is not coming from
+        // the hidden section (it is a swap)
+        [
+          ...selectedColumns.slice(0, dragResult.source.index),
+          ...selectedColumns.slice(dragResult.source.index + 1),
+        ]
+      : selectedColumns;
+
+    const columns = isSwap ? selectedColumns : availableColumns;
+    const draggedColumn = columns.find(
+      (col) => col.key === dragResult.draggableId
+    );
+
+    // push the column in the new position
+    onUpdateColumns([
+      ...items.slice(0, dragResult.destination.index),
+      draggedColumn,
+      ...items.slice(dragResult.destination.index),
+    ]);
+  }
+};
+
+export const ColumnManager = (props) => {
+  const intl = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragStart = () => {
+    setIsDragging(true);
+  };
+
+  const hiddenColumns = differenceWith(
+    props.availableColumns,
+    props.selectedColumns,
+    (a, b) => a.key === b.key
+  );
+
+  const selectInputComponents = {
+    Option: () => null,
+    Menu: () => null,
+    // eslint-disable-next-line react/display-name
+    DropdownIndicator: () => (
+      <Spacings.Inline alignItems="center">
+        <SearchIcon scale="medium" color="primary" />
+      </Spacings.Inline>
+    ),
+  };
+
+  return (
+    <CollapsibleMotion isDefaultClosed={false}>
+      {({ registerContentNode, containerStyles }) => (
+        <Container {...filterDataAttributes(props)}>
+          <Spacings.Stack scale="xs">
+            <HeaderContainer>
+              <Text.Headline as="h3" intlMessage={messages.title} />
+              <AccessibleButton
+                onClick={props.onClose}
+                label={intl.formatMessage(messages.closeButtonLabel)}
+              >
+                <CloseIcon size="medium" />
+              </AccessibleButton>
+            </HeaderContainer>
+            <Spacings.Stack scale="s">
+              <Spacings.Inset scale="xs">
+                <div style={containerStyles}>
+                  <div ref={registerContentNode}>
+                    <DragDropContext
+                      onDragEnd={(dragResult) =>
```

Comment:
Nit: This callback function can be assigned to a variable, wrapped with `React.useCallback`